internal/broker/broker.go

```diff
@@ -725,7 +725,7 @@ func (b *Broker) handleIsAuthenticated(ctx context.Context, session *session, au
 		}
 
 		// If device registration is enabled, ensure that the device is registered.
-		if b.provider.SupportsDeviceRegistration() && b.cfg.registerDevice {
+		if b.provider.SupportsDeviceRegistration() && !session.isOffline && b.cfg.registerDevice {
```

Comment:
Shouldn't we return with an error and handle it in a specific way here?

Reply:
no, I think we want to allow offline login even if device registration is enabled